state/open.go

```diff
@@ -248,10 +257,32 @@ func Initialize(args InitializeParams) (_ *State, err error) {
 		return nil, errors.Annotate(err, "validating initialization args")
 	}
 
-	// When creating the controller model, the new model
-	// UUID is also used as the controller UUID.
 	modelTag := names.NewModelTag(args.ControllerModelArgs.Config.UUID())
-	st, err := open(modelTag, args.MongoInfo, args.MongoDialOpts, args.NewPolicy, args.Clock, nil)
+	if !names.IsValidModel(modelTag.Id()) {
```

Comment:
Not sure since you would expect the uuid to be validated elsewhere. But the code had this check already so I kept it.